reconciler/reconciler.go

```diff
@@ -669,37 +753,85 @@ func (r *Reconciler) inactiveAccountQueue(
 	return nil
 }
 
+func (r *Reconciler) updateLastChecked(index int64) {
+	// Update the lastIndexChecked value if the block
+	// index is greater. We don't acquire the lock
+	// to make this check to improve performance.
+	if index > r.lastIndexChecked {
+		r.lastIndexMutex.Lock()
+
+		// In the time since making the check and acquiring
+		// the lock, the lastIndexChecked could've increased
+		// so we check it again.
+		if index > r.lastIndexChecked {
+			r.lastIndexChecked = index
+		}
+
+		r.lastIndexMutex.Unlock()
+	}
+}
+
 // reconcileActiveAccounts selects an account
 // from the Reconciler account queue and
 // reconciles the balance. This is useful
 // for detecting if balance changes in operations
 // were correct.
-func (r *Reconciler) reconcileActiveAccounts(
-	ctx context.Context,
-) error {
+func (r *Reconciler) reconcileActiveAccounts(ctx context.Context) error { // nolint:gocognit
 	for {
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
 		case balanceChange := <-r.changeQueue:
 			if balanceChange.Block.Index < r.highWaterMark {
+				r.debugLog(
+					"waiting to continue active reconciliation until reaching high water mark...",
+				)
+
+				if err := r.handler.ReconciliationSkipped(
+					ctx,
+					ActiveReconciliation,
+					balanceChange.Account,
+					balanceChange.Currency,
+					HeadBehind,
+				); err != nil {
+					return err
+				}
+
 				continue
 			}
 
+			// TODO: Skip reconciliation if account has been updated so we don't lookup
+			// balance unnecessarily
+
 			amount, block, err := r.bestLiveBalance(
 				ctx,
 				balanceChange.Account,
 				balanceChange.Currency,
 				balanceChange.Block,
 			)
 			if errors.Is(err, ErrBlockGone) {
+				r.debugLog(
+					"block %s gone",
+					types.PrintStruct(balanceChange.Block),
+				)
+
+				if err := r.handler.ReconciliationSkipped(
+					ctx,
+					ActiveReconciliation,
+					balanceChange.Account,
+					balanceChange.Currency,
+					BlockGone,
+				); err != nil {
+					return err
+				}
+
 				continue
 			}
 			if err != nil {
 				// Ensure we don't leak reconciliations if
 				// context is canceled.
 				if errors.Is(err, context.Canceled) {
-					r.wrappedActiveEnqueue(balanceChange)
+					_ = r.wrappedActiveEnqueue(ctx, balanceChange)
```

Comment:
Looks like the error returned by `wrappedActiveEnqueue` is not actionable. Thoughts on at least logging it instead of skipping?